lib/codecs/src/decoding/format/json_paths.rs

```diff
@@ -0,0 +1,1182 @@
+                let current_path_owned = state.current_path().to_owned();
+                let is_exploded = state.explode_path.as_ref().map(|s| s.as_str()) == Some(current_path_owned.as_str());
+
+                if let Some(ValueBuilder::Array(arr)) = state.value_stack.pop() {
+                    if is_exploded {
+                        state.explode_path = None;
+                    } else {
+                        let value = Value::Array(arr);
+                        self.handle_value(state, value)?;
+                    }
+                }
+
+                if should_pop_path {
+                    state.pop_path();
+                }
+            }
+            JsonEvent::FieldName => {
+                let field_name = parser
+                    .current_str()
+                    .map_err(|e| format!("Error reading field name: {:?}", e))?
+                    .to_string();
+                state.push_path(field_name);
+            }
+            JsonEvent::ValueString => {
+                let s = parser
+                    .current_str()
+                    .map_err(|e| format!("Error reading string: {:?}", e))?
+                    .to_string();
+                let value = Value::from(s);
+                self.handle_value_and_pop_if_in_object(state, value)?;
+            }
+            JsonEvent::ValueInt => {
+                let i: i64 = parser
+                    .current_int()
+                    .map_err(|e| format!("Error reading int: {:?}", e))?;
+                let value = Value::from(i);
+                self.handle_value_and_pop_if_in_object(state, value)?;
+            }
+            JsonEvent::ValueFloat => {
+                let f: f64 = parser
+                    .current_float()
+                    .map_err(|e| format!("Error reading float: {:?}", e))?;
+                let value = Value::from(f);
+                self.handle_value_and_pop_if_in_object(state, value)?;
+            }
+            JsonEvent::ValueTrue => {
+                self.handle_value_and_pop_if_in_object(state, Value::from(true))?;
+            }
+            JsonEvent::ValueFalse => {
+                self.handle_value_and_pop_if_in_object(state, Value::from(false))?;
+            }
+            JsonEvent::ValueNull => {
+                self.handle_value_and_pop_if_in_object(state, Value::Null)?;
+            }
+            JsonEvent::NeedMoreInput => {}
+        }
+        Ok(())
+    }
+
+    fn handle_value(&self, state: &mut ParserState, value: Value) -> Result<(), String> {
+        let current_path = state.current_path();
+        let operation = self.path_map.get(&current_path).copied();
+
+        let inside_exploded_array = state.explode_path.as_ref().map(|s| s.as_str()) == Some(current_path.as_str())
+            && matches!(state.value_stack.last(), Some(ValueBuilder::Array(_)));
+
+        if inside_exploded_array {
+            state.events.push((current_path.clone(), value.clone()));
```

Comment:
create Event directly and change `state.events` to SmallVec and then simply pass it on as parsed data?